File: Frontend/RecipeRhapsody/src/app/Models/recipe.ts
import { Ingredient } from './ingredient';
import { PrepTime } from './prepTime';
import { Step } from './step';

export class Recipe {
  createdBy!: string;
  updatedOn?: Date;
  imgUrl!: string;
  category!: string;
  title!: string;
  description!: string;
  ingredients!: Ingredient[];
  steps!: Step[];
  prepTime!: PrepTime[];

  constructor(
    createdBy: string,
    updatedOn: Date,
    imgUrl: string,
    category: string,
    title: string,
    description: string,
    ingredients: Ingredient[],
    steps: Step[],
    prepTime: PrepTime[]
  ) {
    this.imgUrl = imgUrl;
    this.category = category;
    this.title = title;
    this.description = description;
    this.ingredients = ingredients;
    this.steps = steps;
    this.prepTime = prepTime;
    this.createdBy = createdBy;
    this.updatedOn = updatedOn;
  }
}

File: RecipeRhapsody.Domain/Entities/Recipe.cs
namespace RecipeRhapsody.Domain.Entities;

public class Recipe : IUserBase
{
    public int Id { get; set; }
    public ApplicationUser User { get; set; }
    public string UserId { get; set; }

    public DateTime? UpdatedOn { get; set; }
    public string? ImageUrl { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public int Servings { get; set; }
    public string? ServingsYield { get; set; }

    public IEnumerable<Ingredient> Ingredients { get; set; }
    public IEnumerable<Step> Steps { get; set; }
    public IEnumerable<PrepTime> PrepTimes { get; set; }
}
